### PR TITLE
Update `datasets` to 2.16.1

### PR DESCRIPTION
**Pull request by lhoestq**
...and hfh to 0.20.1

This enables:

- Webdataset support for the Dataset Viewer
- Faster data files resolution, which makes listing configs, splits and generating parquet export faster
- Less pressure on the HF back-end (less /tree calls on the Hub, and they no longer use expand=true)

In addition to the update:

- I removed the #2001 hot-fix and replaced it with a proper use of `trust_remote_code`
- I set `HF_DATASETS_TRUST_REMOTE_CODE=0` by default
- I set `datasets.config.USE_PARQUET_EXPORT = False` since we don't need to query parquer exports when loading datasets (this is just to avoid unnecessary calls)

Fix https://github.com/huggingface/datasets-server/issues/2229

TODO:

- [x] fix tests

TODO after the update + deploy:

- [ ] re-generate webdatasets on the hub, e.g.
  - [ ] https://huggingface.co/datasets/timm/imagenet-12k-wds
  - [ ] https://huggingface.co/datasets/pixparse/cc12m-wds
  - [ ] https://huggingface.co/datasets/pixparse/cc3m-wds

### CODE REVIEW

**Comment by lhoestq** [comment]
Not sure why I'm getting auth errors in the e2e tests :/
Cc @severo maybe ?

**Comment by severo** [comment]
Looking at it

**Comment by severo** [comment]
I think it's this: https://github.com/huggingface/datasets-server/pull/2233

**Comment by severo** [comment]
and, even if the token is old, the value in the database was different and never changed, before https://github.com/huggingface/moon-landing/commit/aab4a9b05562e1256c902552fdfb5eed78a21f03 (which was used yesterday to delete the user sessions collection on hub-ci, which replaced the value with the correct one)

cc @coyotte508 for visibility

File: libs/libcommon/src/libcommon/__init__.py [hunk 1]
```diff
@@ -1,2 +1,6 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright 2022 The HuggingFace Authors.
+
+from datasets import config as _datasets_config
+
+_datasets_config.USE_PARQUET_EXPORT = False
```
Comment (severo) [inline libs/libcommon/src/libcommon/__init__.py:6]: maybe add your comment in the code, to understand why we set it to False

**Review by severo — APPROVED** [review]
YEAH

**Comment by lhoestq** [comment]
Still got an e2e test fail on `main` because of auth: https://github.com/huggingface/datasets-server/actions/runs/7420424182/job/20191781984

re-running...

**Pull request merged**